M2/Macaulay2/m2/galois.m2

```diff
@@ -117,7 +117,9 @@ findGalois(ZZ,ZZ) := RingElement => opts -> (p,n) -> (
      else error (opts.Strategy | " is not a valid argument for' Strategy' option")
      )
 
-GF(ZZ,ZZ) := GaloisField => opts -> (p,n) -> (
+GF(ZZ,ZZ) := GaloisField => (
+    memo := new CacheTable;
```

Comment:
Why has this been memo-ized?

Reply:
I forget why this ended up in this particular PR , but GF currently creates a new object every time:

```m2
i1 : tally apply(20, i -> GF 2)

o1 = Tally{GF 2 => 1}
           GF 2 => 1
           GF 2 => 1
           GF 2 => 1
           GF 2 => 1
           GF 2 => 1
           GF 2 => 1
           GF 2 => 1
           GF 2 => 1
           GF 2 => 1
           GF 2 => 1
           GF 2 => 1
           GF 2 => 1
           GF 2 => 1
           GF 2 => 1
           GF 2 => 1
           GF 2 => 1
           GF 2 => 1
           GF 2 => 1
           GF 2 => 1

o1 : Tally
```

Compare to:

```m2
i1 : tally apply(20, i -> ZZ/2)

           ZZ
o1 = Tally{-- => 20}
            2

o1 : Tally
```

Reply:
Yes, for rings, that is the expected behavior.  Otherwise, we have to keep pointers to all of these fields, and therefore they will never be garbage collected.  I guess currently, we are doing something similar for ZZ/p though, so those are likely not being garbage collected either.  In reality, we should have GF(p) be identical to ZZ/p, but it currently isn't...

As these elements are in different rings, that behavior is expected.  Here is what is the intended way to use rings (create it once):
```m2
kk = GF 8
tally apply(20, i -> random kk)
```

Reply:
Ok -- I'll remove that commit.

According to the commit message, this is why I made the change:  https://macaulay2.com/doc/Macaulay2/share/doc/Macaulay2/Macaulay2Doc/html/_random_lp__Type_rp.html

I guess the `tally for i to 100 list random GF 11` should be changed to something like `kk = GF 11; tally for i too 00 list random kk`?

Reply:
Yes, that would be a good change!

Reply:
This was requested before in #1993.